core/ibft_test.go

```diff
@@ -2821,18 +2821,23 @@ func TestIBFT_RunSequence_NewProposal(t *testing.T) {
 	defer cancelFn()
 
 	var (
-		proposal = []byte("proposal")
-		round    = uint64(10)
-		height   = uint64(1)
-		quorum   = uint64(4)
+		proposal                  = []byte("proposal")
+		round                     = uint64(10)
+		height                    = uint64(1)
+		quorum                    = uint64(4)
+		prepareMessageMulticasted = false
 
 		log     = mockLogger{}
 		backend = mockBackend{
 			quorumFn: func(_ uint64) uint64 {
 				return quorum
 			},
 		}
-		transport = mockTransport{}
+		transport = mockTransport{
+			func(_ *proto.Message) {
+				prepareMessageMulticasted = true
+			},
```

Comment:
Why not check the type of message here?

Reply:
Added check

[54b47ba](https://github.com/madz-lab/go-ibft/pull/10/commits/54b47ba58cd987ec9b670512fc99c7a1cfb596d9)